test/integration/dockerregistryclient_test.go

```diff
@@ -21,40 +21,53 @@ const (
 	retryAfter    = time.Millisecond * 500
 )
 
-// retryWhenUnreachable invokes given function several times until it succeeds,
-// returns network unrelated error or maximum number of attempts is reached.
-// Should be used to wrap calls to remote registry to prevent test failures
-// because of short-term outages.
-func retryWhenUnreachable(t *testing.T, f func() error) error {
+var (
+	// Below are lists of error patterns for use with `retryOnErrors` utility.
+
+	// unreachableErrorPatterns will match following error examples:
+	//   Get https://registry.com/v2/: dial tcp registry.com:443: i/o timeout
+	//   Get https://registry.com/v2/: dial tcp: lookup registry.com: no such host
+	//   Get https://registry.com/v2/: dial tcp registry.com:443: getsockopt: connection refused
+	//   Get https://registry.com/v2/: read tcp 127.0.0.1:39849->registry.com:443: read: connection reset by peer
+	//   Get https://registry.com/v2/: net/http: request cancelled while waiting for connection
+	//   Get https://registry.com/v2/: net/http: TLS handshake timeout
+	//   the registry "https://registry.com/v2/" could not be reached
+	unreachableErrorPatterns = []string{
+		"dial tcp",
+		"read tcp",
+		"net/http",
+		"could not be reached",
+	}
+
+	// imageNotFoundErrorPatterns will match following error examples:
+	//   the image "..." in repository "..." was not found and may have been deleted
+	imageNotFoundErrorPatterns = []string{
```

Comment:
Add to the comment to make it clear that this is only ok to do if you're connecting to a non-internal registry.